usr/lib/cca_stdll/cca_specific.c

```diff
@@ -1090,6 +1092,231 @@ CK_RV token_specific_rsa_decrypt(STDLL_TokData_t * tokdata,
     return CKR_OK;
 }
 
+CK_RV token_specific_rsa_oaep_encrypt(STDLL_TokData_t *tokdata,
+                                      ENCR_DECR_CONTEXT *ctx,
+                                      CK_BYTE *in_data,
+                                      CK_ULONG in_data_len,
+                                      CK_BYTE *out_data,
+                                      CK_ULONG *out_data_len,
+                                      CK_BYTE *hash,
+                                      CK_ULONG hlen)
+{
+    CK_RSA_PKCS_OAEP_PARAMS *oaep;
+    long return_code, reason_code, rule_array_count, data_structure_length;
+    unsigned char rule_array[CCA_RULE_ARRAY_SIZE] = { 0, };
+    CK_ATTRIBUTE *attr;
+    OBJECT *key_obj = NULL;
+    CK_RV rc;
+
+    UNUSED(tokdata);
+    UNUSED(hash);
+    UNUSED(hlen);
+
+    rc = object_mgr_find_in_map1(tokdata, ctx->key, &key_obj, READ_LOCK);
+    if (rc != CKR_OK) {
+        TRACE_DEVEL("object_mgr_find_in_map1 failed\n");
+        goto done;
```

Comment:
return rc ?
instead of going down and doing a object_put()

Reply:
see same comment on other commit.